scripts/odbpy/io_place.py

```diff
@@ -287,25 +310,34 @@ def io_place(
     bterm_regex_map = {}
     for side in pin_placement_cfg:
         for regex in pin_placement_cfg[side]:  # going through them in order
-            regex += "$"  # anchor
-            for bterm in bterms:
-                # if a pin name matches multiple regexes, their order will be
-                # arbitrary. More refinement requires more strict regexes (or just
-                # the exact pin name).
-                pin_name = bterm.getName()
-                if re.match(regex, pin_name) is not None:
-                    if bterm in bterm_regex_map:
-                        print(
-                            "Error: Multiple regexes matched",
-                            pin_name,
-                            ". Those are",
-                            bterm_regex_map[bterm],
-                            "and",
-                            regex,
-                        )
-                        sys.exit(os.EX_DATAERR)
-                    bterm_regex_map[bterm] = regex
-                    pin_placement[side].append(bterm)  # to maintain the order
+            if regex[0] == "$": # Sign of Virtual Pins
+                if regex.split("$")[1].isdigit(): # Check that only numbers are followed by the dollar sign
+                    # Convert the number of virtual pins to integer and increment it to the pin_placement list
+                    virtual_pins_count = int(regex.split("$")[1])
+                    pin_placement[side].append(virtual_pins_count)
+                else: # Dollar sign followed by characters other than numbers
+                    print("You provided invalid values for virtual pins")
+                    sys.exit(1)
+            else:
+                regex += "$"  # anchor
+                for bterm in bterms:
+                    # if a pin name matches multiple regexes, their order will be
+                    # arbitrary. More refinement requires more strict regexes (or just
+                    # the exact pin name).
+                    pin_name = bterm.getName()
+                    if re.match(regex, pin_name) is not None:
+                        if bterm in bterm_regex_map:
+                            print(
```

Comment:
Please use a formatted string.

Reply:
I'm not sure what you need here this peace of code was already there in the script, nothing changed by me here

Reply:
Done.